modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -67,7 +73,7 @@ def execute cmd, msg=nil, error_only=false
         std_err.close unless std_in.nil?
       end
       report msg, response, error_only
-      response
+      response if return_result
```

Comment:
Foreman assumes that all calls return json. Empty body is ok though. And yep, good tests are good.